Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp

```diff
@@ -3253,6 +3274,13 @@ void GraphicsLayerCA::updateReplicatedLayers()
         m_layer->insertSublayer(*replicaRoot, 0);
 }
 
+#if HAVE(HDR_SUPPORT)
+void GraphicsLayerCA::updateDrawsHDRContent()
+{
+    m_layer->setContentsFormat(screenContentsFormat(nullptr, this));
```

Comment:
`screenContentsFormat` is a misnomer; it's not the contents format used by the "screen", it just the contents format we've chosen for HDR.

I'd still like us to separate "is showing HDR" from "contents format".

Reply:
I created https://github.com/WebKit/WebKit/pull/40225 to address this issue.